FreeRTOS/FreeRTOS-Kernel
# ARMv8.1-M: Add task dedicated PAC key support

Description
-----------


To harden the security, each task is assigned with a dedicated PAC key, so that attackers needs to get the all the PAC keys right to exploit the system using Return Oriented Programming. The kernel is updated to support the following:

* A PAC key set with a user defined random number generator is generated and is pushed onto the task's stack when a task is created.
* As part of scheduling, the task's PAC key is stacked/unstacked to/from the task's stack when a task is unscheduled/scheduled from/to run.

This [commit](https://github.com/FreeRTOS/FreeRTOS-Kernel/pull/1195/commits/f2cdc150e06ef2457b7dd077521422ac0c5414d4) is an improvement for [MR!1216](https://github.com/FreeRTOS/FreeRTOS-Kernel/pull/1216/) to support MPU port variants stack overflow check.

Test Steps
-----------


Checklist:
----------



Related Issue
-----------


Blocks https://github.com/FreeRTOS/FreeRTOS-Partner-Supported-Demos/pull/19

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
